torchgen/api/python.py

```diff
@@ -951,6 +951,9 @@ def argument_type_str_pyi(t: Type) -> str:
             )
         elif str(t.elem) == "float":
             ret = "Sequence[_float]"
+        elif str(t.elem) == "SymInt" and t.size == 1:
+            elem = argument_type_str_pyi(t.elem)
+            ret = f"Union[{elem}, Sequence[{elem}]]"
```

Comment:
Non-list SymInts are handled like this:

```py
        if t.name == BaseTy.SymInt:
            ret = "Union[_int, SymInt]"
```

We have this rule for ints that are in lists:

```py
    elif isinstance(t, ListType):
        if str(t.elem) == "int":
            ret = "Union[_int, _size]" if t.size is not None else "_size"
```

I don't quite understand it, is it even correct? So I went with a more conservative rule for SymInts.

New signature of roll. It has two Unions, but I think it's better than inlining and hardcoding in torchgen.

```py
def roll(input: Tensor, shifts: Union[Union[_int, SymInt], Sequence[Union[_int, SymInt]]], dims: Union[_int, _size] = ()) -> Tensor: ...
```